pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -213,87 +206,101 @@ def __init__(
-                return processing_fn(transformed_tapes)
-            return transformed_tapes, processing_fn
+            @qml.transform
+            def printer(tape):
+                print("I have a tape: ", tape)
+                return (tape, ), lambda x: x[0]
 
-        if isinstance(obj, qml.workflow.QNode):
-            if capture.enabled():
-                new_qnode = self.default_qnode_transform(obj, targs, tkwargs)
-                return self._capture_callable_transform(new_qnode, targs, tkwargs)
-            return self._qnode_transform(obj, targs, tkwargs)
+            @printer.register
+            def _(obj: qml.operation.Operator, *targs, **tkwargs):
+                print("I have an operator:", obj)
+                return obj
 
-        if isinstance(obj, qml.devices.Device):
-            return self._device_transform(obj, targs, tkwargs)
+        >>> printer(qml.X(0))
+        I have an operator: X(0)
+        X(0)
 
-        if obj.__class__.__name__ == "QJIT":
-            raise TransformError(
-                "Functions that are wrapped / decorated with qjit cannot subsequently be"
-                f" transformed with a PennyLane transform (attempted {self})."
-                f" For the desired affect, ensure that qjit is applied after {self}."
-            )
 
-        if callable(obj):
-            if capture.enabled():
-                return self._capture_callable_transform(obj, targs, tkwargs)
-            return self._qfunc_transform(obj, targs, tkwargs)
+        """
+        return self._apply_transform.register
 
-        if isinstance(obj, Sequence) and all(isinstance(q, qml.tape.QuantumScript) for q in obj):
-            return self._batch_transform(obj, targs, tkwargs)
+    def generic_apply_transform(self, obj, *targs, **tkwargs):
+        """generic_apply_transform(obj, *targs, **tkwargs)
+        Generic application of a transform that forms the default for all transforms.
 
-        # Input is not a QNode nor a quantum tape nor a device.
-        # Assume Python decorator syntax:
-        #
-        # result = some_transform(*transform_args)(qnode)(*qnode_args)
+        Args:
+            obj: The object we want to transform
+            *targs: The arguments for the transform
+            **tkwargs: The keyword arguments for the transform.
 
-        raise TransformError(
-            "Decorating a QNode with @transform_fn(**transform_kwargs) has been "
-            "removed. Please decorate with @functools.partial(transform_fn, **transform_kwargs) "
-            "instead, or call the transform directly using qnode = transform_fn(qnode, "
-            "**transform_kwargs). Visit the deprecations page for more details: "
-            "https://docs.pennylane.ai/en/stable/development/deprecations.html#completed-deprecation-cycles",
-        )
+        """
+        return generic_apply_transform(obj, self, *targs, **tkwargs)
+
+    @classmethod
+    def generic_register(cls, arg=None):
```

Comment:
Should this be a `staticmethod` instead? I'm not sure what makes more sense here.